contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -720,4 +739,11 @@ contract StakeManager is IStakeManager {
         validatorState[epoch].stakerCount += stakerCount;
     }
 
+    function pubToAddress(bytes memory pub) public pure returns (address addr) {
+        bytes32 hash = keccak256(pub);
+        assembly {
+            mstore(0, hash)
```

Comment:
Shouldn't we use the free memory pointer at `0x40` ?

Reply:
We must check if the `pub`'s length is 64 otherwise it will give a different address accidently.

Reply:
We must check `pub`'s length to 64 to avoid accidental address as signer.

Reply:
> Shouldn't we use the free memory pointer at 0x40 ?

Might not be required. Solidity docs say that `0x00 - 0x3f (64 bytes): scratch space for hashing methods`. Not sure what it means, please verify :) https://solidity.readthedocs.io/en/v0.5.12/miscellaneous.html#layout-in-memory

Reply:
yes saw that, and went back to type casting.